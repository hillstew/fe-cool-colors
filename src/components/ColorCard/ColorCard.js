import React from 'react';

export const ColorCard = ({ color, toggleLocked }) => {
  const { hex, isLocked, id } = color;

  return (
    <div className="ColorCard" style={{ backgroundColor: hex }}>
      <p className="hex-code">{hex}</p>
      <button
        onClick={() => toggleLocked(id)}
        className={isLocked ? 'locked-button' : 'unlocked-button'}
      />
    </div>
  );
};
