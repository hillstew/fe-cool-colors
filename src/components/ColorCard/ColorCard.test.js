import React from 'react';
import { shallow } from 'enzyme';
import { ColorCard } from './ColorCard';

describe('ColorCard', () => {
  let wrapper;
  let mockColor;
  let mockToggleLocked;

  beforeEach(() => {
    mockColor = { hex: '#2309ED', id: 23, isLocked: false };
    mockToggleLocked = jest.fn();
    wrapper = shallow(
      <ColorCard color={mockColor} toggleLocked={mockToggleLocked} />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
