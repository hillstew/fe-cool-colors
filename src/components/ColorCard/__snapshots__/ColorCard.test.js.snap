// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorCard should match snapshot 1`] = `
<div
  className="ColorCard"
  style={
    Object {
      "backgroundColor": "#2309ED",
    }
  }
>
  <p
    className="hex-code"
  >
    #2309ED
  </p>
  <button
    className="unlocked-button"
    onClick={[Function]}
  />
</div>
`;
